cpp/include/cuspatial/vec_2d.hpp

```diff
@@ -98,6 +107,35 @@ vec_2d<T> CUSPATIAL_HOST_DEVICE operator*(T const& r, vec_2d<T> vec)
   return vec * r;
 }
 
+/**
+ * @brief Total order two points on plane
+ */
+template <typename T>
+bool CUSPATIAL_HOST_DEVICE operator<(vec_2d<T> const& lhs, vec_2d<T> const& rhs)
+{
+  if (lhs.x < rhs.x)
```

Comment:
Should probably document the ordering, since X taking priority over Y is arbitrary.